src/Resources/Resources/PrivateLinks/GetAzureResourceManagementPrivateLink.cs

```diff
@@ -0,0 +1,79 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Linq;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.Resources.Models.PrivateLinks;
+using Microsoft.Azure.Commands.Resources.PrivateLinks.Common;
+using Microsoft.Azure.Management.ResourceManager;
+
+namespace Microsoft.Azure.Commands.Resources.PrivateLinks
+{
+    /// <summary>
+    /// Get-AzResourceManagementPrivateLink Cmdlet
+    /// </summary>
+    [Cmdlet(
+        VerbsCommon.Get,
+        ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "ResourceManagementPrivateLink",
+        DefaultParameterSetName = Constants.ParameterSetNames.GetParameterSet,
+        SupportsShouldProcess = true), OutputType(typeof(PSResourceManagementPrivateLink))]
+    public class GetAzureResourceManagementPrivateLink : PrivateLinksCmdletBase
+    {
+        #region Cmdlet Parameters and Parameter Set Definitions
+
+        [Parameter(
+            ParameterSetName = Constants.ParameterSetNames.GetParameterSet,
+            Mandatory = false,
+            HelpMessage = Constants.HelpMessages.ResourceGroupName,
+            Position = 0)]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; } = null;
+
+        [Parameter(
+            ParameterSetName = Constants.ParameterSetNames.GetParameterSet,
+            Mandatory = false,
+            HelpMessage = Constants.HelpMessages.PrivateLinkName,
+            Position = 1)]
+        [ValidateNotNullOrEmpty]
+        public string PrivateLinkName { get; set; }
+        #endregion
+
+        public override void ExecuteCmdlet()
+        {
+            try
+            {
+                if (!string.IsNullOrEmpty(ResourceGroupName) && !string.IsNullOrEmpty(PrivateLinkName))
+                {
+                    var response = ResourceManagementPrivateLinkClient.ResourceManagementPrivateLink.Get(
+                        resourceGroupName: ResourceGroupName,
+                        rmplName: PrivateLinkName);
+                    WriteObject(new PSResourceManagementPrivateLink(response));
+                }
+                else if (string.IsNullOrEmpty(ResourceGroupName) && string.IsNullOrEmpty(PrivateLinkName))
+                {
+                    //List all the private links in a subscription no parameters needed
+                    var response = ResourceManagementPrivateLinkClient.ResourceManagementPrivateLink.List();
+                    var items = response.Value.Select(resourceManagementPrivateLink => new PSResourceManagementPrivateLink(resourceManagementPrivateLink))
+                        .ToList();
+                    WriteObject(items);
+                }
```

Comment:
How about the scenario that only resource group name is provided and PrivateLinkName is absent?